components/server/src/database/measurements.py

```diff
@@ -1,27 +1,29 @@
 """Measurements collection."""
 
 from datetime import datetime, timedelta
-from typing import Optional, cast
+from typing import Optional
 
 import pymongo
 from pymongo.database import Database
 
+from model.measurement import Measurement
 from model.metric import Metric
-from model.queries import get_attribute_type, get_measured_attribute
-from server_utilities.functions import iso_timestamp, percentage
-from server_utilities.type import MeasurementId, MetricId, Scale, Status, TargetType
+from server_utilities.functions import iso_timestamp
+from server_utilities.type import MeasurementId, MetricId
 
 
-def latest_measurement(database: Database, metric_uuid: MetricId):
+def latest_measurement(database: Database, metric_uuid: MetricId, metric: Metric) -> Optional[Measurement]:
```

Comment:
Could the uuid be part of the metric instance instead of a separate input parameter to this and other functions? That would clean up the interface.

Reply:
Nice idea: Metric.latest_measurement() and Metric.latest_successful_measurement(). Disadvantage is that it would mix the model layer (Metric) with the database layer.

Reply:
What is exactly the thought behind the database layer? Looking through it, my intuition would indeed be to move basically everything that is currently in there onto the different models. But that might just be me being too used to orm patterns :D

Reply:
I prefer model classes to have no knowledge of storage or presentation, cf. "clean architecture". See e.g. https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html